go/database/mpt/proof_test.go

```diff
@@ -0,0 +1,462 @@
+			mock: func(mock *MockNodeSource) {
+				var account Node = &AccountNode{address: common.Address{0xA}}
+				gomock.InOrder(
+					mock.EXPECT().getViewAccess(gomock.Any()).Return(shared.MakeShared(account).GetViewHandle(), nil),
+					mock.EXPECT().getViewAccess(gomock.Any()).Return(shared.MakeShared(node).GetViewHandle(), injectedErr),
+				)
+			},
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			nodeSource := NewMockNodeSource(ctrl)
+			nodeSource.EXPECT().getConfig().AnyTimes().Return(S5LiveConfig)
+			mockPathHashing(nodeSource)
+
+			test.mock(nodeSource)
+			root := NewNodeReference(EmptyId())
+			source := NewProofSource(&root, nodeSource)
+
+			if _, err := source.CreateStorageProofs(common.Address{0xA}, common.Key{0x1}); !errors.Is(err, injectedErr) {
+				t.Errorf("getting proof should fail")
+			}
+		})
+	}
+}
+
+func TestCreateAccountProofs_FailingNodeSources(t *testing.T) {
+	ctrl := gomock.NewController(t)
+
+	injectedErr := fmt.Errorf("injected error")
+
+	nodeSource := NewMockNodeSource(ctrl)
+	nodeSource.EXPECT().getConfig().AnyTimes().Return(S5LiveConfig)
+	mockPathHashing(nodeSource)
+
+	var account Node = &AccountNode{address: common.Address{0xA}}
+	nodeSource.EXPECT().getViewAccess(gomock.Any()).Return(shared.MakeShared(account).GetViewHandle(), injectedErr)
+
+	root := NewNodeReference(EmptyId())
+	source := NewProofSource(&root, nodeSource)
+
+	if _, err := source.CreateAccountProof(common.Address{0xA}); !errors.Is(err, injectedErr) {
+		t.Errorf("getting proof should fail")
+	}
+}
+
+func TestCreateStorageProofs_HasherFails(t *testing.T) {
+	ctrl := gomock.NewController(t)
+
+	injectedErr := fmt.Errorf("injected error")
+
+	mockHasher := NewMockhasher(ctrl)
+	mockHasher.EXPECT().encode(gomock.Any(), gomock.Any(), gomock.Any()).Return([]byte{}, injectedErr)
+	mockHasherFunc := func() hasher {
+		return mockHasher
+	}
+
+	nodeSource := NewMockNodeSource(ctrl)
+	nodeSource.EXPECT().getConfig().AnyTimes().Return(MptConfig{Hashing: hashAlgorithm{createHasher: mockHasherFunc}})
+	mockPathHashing(nodeSource)
+
+	var account Node = &AccountNode{address: common.Address{0xA}}
+	nodeSource.EXPECT().getViewAccess(gomock.Any()).Return(shared.MakeShared(account).GetViewHandle(), nil)
+
+	root := NewNodeReference(EmptyId())
+	source := NewProofSource(&root, nodeSource)
+
+	if _, err := source.CreateStorageProofs(common.Address{0xA}, common.Key{0x1}); !errors.Is(err, injectedErr) {
+		t.Errorf("getting proof should fail")
+	}
+}
+
+func mockPathHashing(nodeSource *MockNodeSource) {
+	nodeSource.EXPECT().hashKey(gomock.Any()).AnyTimes().DoAndReturn(func(key common.Key) common.Hash {
+		var h common.Hash
+		copy(h[:], key[:])
+		return h
+	})
+	nodeSource.EXPECT().hashAddress(gomock.Any()).AnyTimes().DoAndReturn(func(address common.Address) common.Hash {
+		var h common.Hash
+		copy(h[:], address[:])
+		return h
+	})
+}
+
+func mockNodeSource(ctrl *gomock.Controller, nodes []Node) NodeSource {
```

Comment:
This name is quite generic and visible in the entire MPT package. Can you think of a more specific name describing what it is providing?